flexmeasures/api/common/schemas/sensors.py

```diff
@@ -0,0 +1,135 @@
+from typing import Union
+
+from marshmallow import fields
+
+from flexmeasures.api import FMValidationError
+from flexmeasures.api.common.utils.api_utils import get_weather_sensor_by
+from flexmeasures.utils.entity_address_utils import (
+    parse_entity_address,
+    EntityAddressException,
+)
+from flexmeasures.data.models.assets import Asset
+from flexmeasures.data.models.markets import Market
+from flexmeasures.data.models.weather import WeatherSensor
+from flexmeasures.data.models.time_series import Sensor
+
+
+class EntityAddressValidationError(FMValidationError):
+    status = "INVALID_DOMAIN"  # USEF error status
+
+
+class SensorField(fields.Str):
+    """Field that deserializes to a Sensor, Asset, Market or WeatherSensor
+    and serializes back to an entity address (string)."""
+
+    # todo: when Actuators also get an entity address, refactor this class to EntityField,
+    #       where an Entity represents anything with an entity address: we currently foresee Sensors and Actuators
```

Comment:
Note that, under the fm1 scheme, persisted objects with an entity address should have a "mutually unique" id (unique within the set of persisted objects with an entity address). Right now, all of those objects are sensors (i.e. either Sensors, or Assets, Markets, WeatherSensors, the three of which we intend to replace by Sensors).

If we are going to persist Actuators, callable through our API, they should get an entity address, too. We'd then need to ensure Sensor ids and Actuator ids are mutually unique (or update the fm scheme again, to distinguish between the two types).